packages/core/src/install/index.ts

```diff
@@ -563,7 +566,9 @@ export function createReadPackageHook (
   if (!isEmpty(overrides ?? {})) {
     hooks.push(createVersionsOverrider(overrides!, lockfileDir))
   }
-  hooks.push(createPackageExtender(fromPairs(compatPackageExtensions)))
+  if (!ignoreCompatibilityDb) {
+    hooks.push(createPackageExtender(fromPairs(compatPackageExtensions)))
+  }
```

Comment:
https://github.com/pnpm/pnpm/issues/5132#issuecomment-1201571452 :

> It's also worth considering whether these fixups should be applied earlier, since `pnpmfile.cjs` is less useful if it is not the final authority about version selections.

(This part could maybe be a separate PR since it doesn't affect our use case with `ignoreCompatibilityDb=true`. But I do suspect that users would be surprised to learn that the input to `pnpmfile.cjs` does not reflect some hooks, and the actions of `pnpmfile.cjs` may get overridden by compatibility db fixups.)